src/it/resources-noncompilable/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputWhitespaceAroundArrow.java

```diff
@@ -0,0 +1,100 @@
+package com.google.checkstyle.test.chapter4formatting.rule462horizontalwhitespace;
+
+/** some javadoc. */
+public class InputWhitespaceAroundArrow {
+
+  static { 
+    // violation below ''->' is not preceded with whitespace.'
+    new JCheckBox().addActionListener((final ActionEvent e)-> {
```

Comment:
I thought we always need to add input to exclude list 
src/it/resources-noncompilable/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputWhitespaceAroundArrow.java

If it has violations.

Reply:
in some future we will automate validation on this.
https://github.com/checkstyle/checkstyle/issues/17258